src/core/thread/mle.cpp

```diff
@@ -2772,6 +2758,18 @@ ThreadError Mle::HandleChildIdResponse(const Message &aMessage, const Ip6::Messa
                                                  networkData.GetNetworkData(), networkData.GetLength());
 
     mNetif.GetActiveDataset().ApplyConfiguration();
+
+    // if 'R' bit of SecurityPolicy is cleared, Router-capable Thread 1.x devices SHALL NOT attempt to
+    // register as Active Router or Leader
+    if (!(mNetif.GetKeyManager().GetSecurityPolicyFlags() & OT_SECURITY_POLICY_ROUTERS))
+    {
+        mNetif.GetMle().SetRouterRoleEnabled(false);
```

Comment:
I believe we need to consider updating the router role based on the security flag whenever the Active Operational Dataset changes.